testtools/_deferred.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) testtools developers. See LICENSE for details.
+
+"""Utilities for Deferreds."""
+
+from functools import partial
+
+from testtools.content import TracebackContent
+
+
+class DeferredNotFired(Exception):
+    """Raised when we extract a result from a Deferred that's not fired yet."""
+
+    def __init__(self, deferred):
+        msg = "%r has not fired yet." % (deferred,)
+        super(DeferredNotFired, self).__init__(msg)
```

Comment:
Intentionally not keeping the Deferred as an attribute of the exception?  The repr of a Deferred is usually nearly useless in my experience.  Maybe knowing its id() is useful for correlating it with another occurrence... apart from that it's just noise, though.